### PR TITLE
Add a modified voting app based on registry pull [full ci]

### LINKED ISSUE #2358: Get a modified Voting app working on VIC and update the integration test
The current docker voting app cannot work on VIC right now.  It relies on mapped volumes, network alias, and docker build.  Network alias is currently being fixed, but the other two will not be a part of 1.0/GA.  We need to redesign the voting app to work on VIC.  This will involve building custom images and pushing to a private registry.

### PR DESCRIPTION
**Pull request by sflxn**
Add a modified version of docker's voting app compose yml file that
runs on VIC using docker pull from the global registry hub.

Resolves #2358

### CODE REVIEW

**Comment by sflxn** [comment]
@mhagen-vmware @chengwang86

**Comment by anchal-agrawal** [comment]
LGTM

File: tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.robot [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-    Run Keyword If  '${status}' == 'closed'  Fail  Test 3-2-Docker-Compose-Voting-App.robot needs to be updated now that Issue #2358 has been resolved
-    Log  Issue \#2358 is blocking implementation  WARN
-    #Run  git clone https://github.com/docker/example-voting-app test
-    #${rc}  ${output}=  Run And Return Rc And Output  cd test; DOCKER_HOST=${vch-ip}:2375 docker-compose up
-    #Log  ${output}
-    #Should Be Equal As Integers  ${rc}  0
\ No newline at end of file
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} login --username=victest --password=vmware!123
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}  ${output}=  Run And Return Rc And Output  cd demos/compose/voting-app; COMPOSE_HTTP_TIMEOUT=300 DOCKER_HOST=${vch-ip}:2375 docker-compose up -d
```
Comment (mhagen-vmware) [inline tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.robot]: Run Process  with cwd= would be cleaner. i.e.
${out}=  Run Process  DOCKER_HOST\=tcp://${ip} go test  shell=True  cwd=${CURDIR}/../../demos/compose/voting-app  timeout=10min  on_timeout=kill
Log  ${out.rc}
Log  ${out.stdout}
Log  ${out.stderr}

File: tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.md [hunk 2]
```diff
@@ -12,14 +12,13 @@ To verify that VIC appliance can work when deploying the example docker voting a
...
+1. Deploy VIC appliance to the vSphere server
+2. Log into the docker hub
 3. Issue the following command in the docker voting app folder:  
-```DOCKER_HOST=<VCH IP> docker-compose up``` 
-4. Verify that the server is running on http://localhost:5000 and the results are found on http://localhost:5001
+```cd demos/compose/voting-app; COMPOSE_HTTP_TIMEOUT=300 DOCKER_HOST=<VCH IP> docker-compose up```
 
 #Expected Outcome:
-Docker compose should return with success and the server and results should be running.
+Docker compose should return with success.
```
Comment (mhagen-vmware) [inline tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.md]: Any reason not to verify the output of the app as well?

File: tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.robot (hunk 1 above)
Comment (sflxn) [inline tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.robot]: Will make the CWD change.  My intent on the demos folder is that we will add more examples in there in the future.

File: tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.md (hunk 2 above)
Comment (sflxn) [inline tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.md]: Added in tests with curl.

File: tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.robot [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
-    Log  Issue \#2358 is blocking implementation  WARN
-    #Run  git clone https://github.com/docker/example-voting-app test
-    #${rc}  ${output}=  Run And Return Rc And Output  cd test; DOCKER_HOST=${vch-ip}:2375 docker-compose up
-    #Log  ${output}
-    #Should Be Equal As Integers  ${rc}  0
\ No newline at end of file
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${params} login --username=victest --password=vmware!123
+    Should Be Equal As Integers  ${rc}  0
+    ${out}=  Run Process  docker-compose  up  -d  shell=True  cwd=${CURDIR}/../../../demos/compose/voting-app   env:COMPOSE_HTTP_TIMEOUT=300  env:DOCKER_HOST=${vch-ip}:2375
+    Log  ${out.rc}
```
Comment (mhagen-vmware) [inline tests/test-cases/Group3-Docker-Compose/3-2-Docker-Compose-Voting-App.robot]: Needs to be ${out.rc}, don't need all the log statements above that was just and example of how to use the return structure from run process

**Comment by sflxn** [comment]
ping

**Comment by sflxn** [comment]
ping again

**Comment by sflxn** [comment]
ping

**Comment by mdubya66** [comment]
lgtm

**Pull request merged**